components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1316,6 +1316,36 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
     }
 
+    public async moveWorkspace(ctx: TraceContext, workspaceId: string, targetOrganizationId: string): Promise<void> {
```

Comment:
This is possibly misleading. It moves the workspace, but it's attribution still remains in the previous org. I wonder if we should even support moving workspaces because it creates this ambiguity about who pays for it after it's been moved

Reply:
The attribution is on the sessions in the past. Why do you think we should change that after the fact?
Also, why is it ambiguous who pays?

Reply:
The workspace is attributed the moment it is started, which may belong to org A. I would naturally expect that org A pays for it.

If I now move the workspace to org B, it creates the ambiguity on who actually pays for the workspace (and it's previous instances). Yes, from the code it would always be the org currently selected, but from the user perspective it's not clear and feels like modifying history.

In other words, do we really need to support moving workspaces? Could we instead make it more obvious under which org you're starting a workspace?

Reply:
I should have mentioned in the description that attribution is the reason we are introducing this feature.
I.e. users have created workspaces on the "wrong" org in the past and are not able to start them now, because that org might not have enough credits. They can move their existing workspaces to the org they wish to get unblocked.

Reply:
In that case, what if we added a feature which would be "Start this workspace in another org" rather than moving the workspace? This would create a new workspace, but could use the current (must be stopped) one as the basis. We retain immutability this way.

Reply:
I'm still not sure I understand the issue you see here tbh, let's talk through in our 1on1 today

Reply:
Chatted through it in more detail.

There's an outstanding issue on what happens with workspaces that belong to a project (as a project has project specific env variables, which will also be in snapshots of the workspace).

In general, we agreed that if we can find a way to address the problem without an explicit feature for it (migration), it would be preferable.